presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/test/RuleAssert.java

```diff
@@ -149,15 +150,15 @@ private RuleApplication applyRule()
 
         PlanNode memoRoot = memo.getNode(memo.getRootGroup());
 
-        return rule.withMatch(new PlanNodeMatcher(lookup), memoRoot, (rule, match) -> {
-            if (!rule.isEnabled(session) || match.isEmpty()) {
-                return new RuleApplication(lookup, symbolAllocator.getTypes(), Optional.empty());
-            }
+        PlanNodeMatcher matcher = new PlanNodeMatcher(lookup);
+        Match match = matcher.match(rule.getPattern(), memoRoot);
```

Comment:
Don't use raw types (that's what Martin told me ;)). Once you try not to, it becomes clear why the method was there and what its javadoc was talking about.

I agree that the method was used only here (so far), but it's needed to avoid the raw types here. I'd just move it here.